sonaranalyzer-dotnet/its/regression-test.ps1

```diff
@@ -25,33 +35,121 @@ function Build-Project([string]$ProjectName, [string]$SolutionRelativePath) {
         /fl /flp:"logFile=output\$ProjectName.log;verbosity=d"
 }
 
-function Initialize-ActualFolder {
+function Initialize-ActualFolder() {
+    $timer = [system.diagnostics.stopwatch]::StartNew()
+
     Write-Host "Initializing the actual issues Git repo with the expected ones"
     if (Test-Path .\actual) {
         Remove-Item -Recurse -Force actual
     }
+    # this copies no files if ruleId is not set, and all but ending with ruleId if set
 
-    Copy-Item .\expected -Destination .\actual -Recurse
-    Invoke-InLocation "actual" {
-        Exec { git init }
-        Exec { git config user.email "regression-test@example.com" }
-        Exec { git config user.name "regression-test" }
-        Exec { git add -A . }
-        Exec { git commit -m "Initial commit with expected files" | out-null }
-        Get-ChildItem * -Include *.json -Recurse | Remove-Item -Force
-    }
+    CopyFolderRecursively -from .\expected -to .\actual -exclude "*${ruleId}.json"
+    Write-Host "Initialize-ActualFolder:" $timer.Elapsed.TotalSeconds
 }
 
-function Initialize-OutputFolder {
+function Initialize-OutputFolder() {
+    $timer = [system.diagnostics.stopwatch]::StartNew()
+
     Write-Host "Initializing the output folder"
     if (Test-Path .\output) {
         Remove-Item -Recurse -Force output
     }
     New-Item -ItemType directory -Path .\output | out-null
-    Copy-Item ".\AllSonarAnalyzerRules.ruleset" -Destination ".\output"
+
+    if ($ruleId) {
+        $template = Get-Content -Path ".\SingleRule.ruleset.template" -Raw
+        $rulesetWithOneRule = $template.Replace("<Rule Id=`"$ruleId`" Action=`"None`" />", `
+                                                "<Rule Id=`"$ruleId`" Action=`"Warning`" />")
+        Set-Content -Path ".\output\AllSonarAnalyzerRules.ruleset" -Value $rulesetWithOneRule
+    }
+    else {
+        Copy-Item ".\AllSonarAnalyzerRules.ruleset" -Destination ".\output"
+    }
+    Write-Host "Initialize-OutputFolder:" $timer.Elapsed.TotalSeconds
+}
+
+function Export-AnalyzerPerformancesFromLogs([string[]]$buildLogsPaths) {
+    return $buildLogsPaths |
```

Comment:
[minor][cosmetic] I would put the piping operator on the next line for all calls, it makes things more readable IMO.